### PR TITLE
Proposed code refactor of ph5tostationxml.py to limit repeated code.

### PR DESCRIPTION
**Pull request by nick-iris**
Contained in this pull request is a proposed code refactor of ph5tostationxml.py. This is just a suggestion, so don't feel like you need to merge this PR.

The code has been refactored so that the creation of ObsPy objects are in separate methods (_create_obs_network_, _create_obs_station_, _create_obs_channel_), and some repetitive code (explained below) was moved to separate methods. I personally feel that these changes make it easier to follow the logic of the main Read_Stations method loop, and will make it easier to write unittest.

Some of the changes are outlined below:
The _does_pattern_exists_ method can be used instead of repeating the same code for every patterns list check.
Repetitive checks:
- https://github.com/PIC-IRIS/PH5/blob/master/kitchen/ph5tostationxml.py#L238-L244
- https://github.com/PIC-IRIS/PH5/blob/master/kitchen/ph5tostationxml.py#L342-L347
- https://github.com/PIC-IRIS/PH5/blob/master/kitchen/ph5tostationxml.py#L351-L356
- https://github.com/PIC-IRIS/PH5/blob/master/kitchen/ph5tostationxml.py#L374-L379

Proposed change:
```
def does_pattern_exists(self, patterns_list, other_pattern):
    for pattern in patterns_list:
        if fnmatch.fnmatch(other_pattern, pattern):
            return True
    return False
```

Likewise the addition of the _does_experiment_pattern_exists_ method replaced the following.
Repetitive checks:
- https://github.com/PIC-IRIS/PH5/blob/master/kitchen/ph5tostationxml.py#L522-L529
- https://github.com/PIC-IRIS/PH5/blob/master/kitchen/ph5tostationxml.py#L531-L539

Proposed change:
```
def does_experiment_pattern_exists(self, request_patterns, code):
    l = []
    for pattern in request_patterns:
        if fnmatch.fnmatch(self.experiment_t[0][code], pattern):
            l.append(1)
    if not l:
        return False
    else:
        return True
```

Lastly instead of repeating the same check for the latitude/longitude of each channel and station, I created a separate method that does this.
Repetitive checks:
- https://github.com/PIC-IRIS/PH5/blob/master/kitchen/ph5tostationxml.py#L262-L285
- https://github.com/PIC-IRIS/PH5/blob/master/kitchen/ph5tostationxml.py#L413-L427

Proposed change:
```
def is_lat_lon_match(self, latitude, longitude):   
    if  not -90 <= float(latitude) <= 90:
        return False 
    elif not  -180 <= float(longitude) <= 180:
        return False              
    elif self.args.get('minlat') and float(
            self.args.get('minlat')) > float(latitude):
        return False
    elif self.args.get('minlon') and float(
            self.args.get('minlon')) > float(longitude):
        return False
    elif self.args.get('maxlat') and float(
            self.args.get('maxlat')) < float(latitude):
        return False
    elif self.args.get('maxlon') and float(
            self.args.get('maxlon')) < float(longitude):
        return False
    else:
        return True
```

### CODE REVIEW

File: webservices/ph5tostationxml.py [hunk 1]
```diff
@@ -495,6 +489,20 @@ def get_network_date(self):
...
+            # for station level show the selected_number_of_channels element
+            for station in network.stations:
+                station.selected_number_of_channels = 0
+                station.channels = []
+        elif self.args.get('level').upper() == "CHANNEL":
+            for station in network.stations:
+                for channel in station.channels:
+                    channel.response = None
+        return network
+
```
Comment (nick-iris) [inline webservices/ph5tostationxml.py]: This is not as efficient as checking the level in the main loop, but allows for all the trimming to be performed in one place which is more maintainable.

**Comment by nick-iris** [comment]
I was able to fix issue #25 by having the program always filter to the response level. The program then trims the inventory to the requested visual level before writing out the StationXML/TXT (see [trim_to_level](https://github.com/PIC-IRIS/PH5/blob/b88f3a4b0466434b041e00d73eaff6d0fe97bc0e/kitchen/ph5tostationxml.py#L491-L503) method).

**Comment by derick-hess** [comment]
Thanks on #25. That was basically my plan for 25 as well. I'll get this all reviewed tomorrow

**Comment by nick-iris** [comment]
#16 was also addressed by these changes.

**Pull request merged**